### ISSUE TITLE
Redesign the way the x-axis works for wxLineChart

### ISSUE THREAD

**Issue by xleclercq** [wxIshiko/wxCharts#35]
At the moment it only works if there are as many labels as data points. It should be possible to have data points at other positions than the tick marks. A good example of this is for time series where there can be many points between marks,

**Comment by Anton-V-K** [comment]
I've been investigating the code of `wxLineChart` (and related classes) a bit (while trying to implement #108 ), and it seems a lot of refactoring will be needed to implement the redesign of x-axis (unless I overlooked something to make it simpler).
AFAIK the x-axis supports arbitrary (non-ordered) non-numeric marks (a kind of labels), so it will be tricky to implement interpolation of points when the x-values are not ordered.
Anyway I would pick this issue up, since it is likely it will be needed for my application.
And it would be great if you shared your ideas/hints how to implement it :)

**Comment by xleclercq** [comment]
Yes a lot of refactoring will be needed for this. I need to check where I left on this but if memory serves I was going to define distinct types of x-axis with a common interface so that any kind of thing is possible there. But this needs a lot of refactoring to have a clean interface for the axes.
